src/frontend/parser.mly

```diff
@@ -110,18 +110,21 @@ generated_quantities_block:
   | GENERATEDQUANTITIESBLOCK LBRACE tvds=list(top_vardecl_or_statement) RBRACE
     { grammar_logger "generated_quantities_block" ; tvds }
 
+build_id(ID):
+  | ID { fun id -> (
+           grammar_logger ("identifier " ^ id);
+           {name=id; id_loc=Location_span.of_positions_exn $startpos $endpos})}
```

Comment:
I think this is equivalent to defining a normal ocaml function and calling that. You can define ocaml functions at the top of the file (see `reducearray`). Then you can call that function where you're currently calling `f`, and also in the `id=IDENTIFIER` rule

Reply:
I tried and it didn't have access to `$startpos` and `$endpos`, tho I guess the rest of this would possibly work if we passed those in at each use-site? Would that be better?

Reply:
Yeah, you'd have to pass in $startpos and $endpos. I do think that would be a little bit better